middleware.go

```diff
@@ -35,6 +36,29 @@ func (m *Middleware) isOriginAllowed(origin string) bool {
 		return true
 	} else if m.AllowedOrigins[origin] != nil {
 		return true
+	} else if m.originMatchesRegex(origin) {
+		return true
+	}
+
+	return false
+}
+
+func (m *Middleware) originMatchesRegex(origin string) bool {
+	re, err := regexp.Compile("^/(.+)/$")
+	if err != nil {
+		return false
+	}
+
+	for k, _ := range m.AllowedOrigins {
```

Comment:
should omit 2nd value from range; this loop is equivalent to `for k := range ...`